### DISCUSSION TITLE
Generating pure grpc-web based client without dependency on @grpc/grpc-js

### DISCUSSION THREAD

**Discussion by oc007us** [thesayyn/protoc-gen-ts#248] [Q&A]
Greetings,

I am facing an issue with generating a pure grpc-web service client.

Here is my  protoc command:

protoc --ts_out=./src/gen -I=./proto MyService.proto --ts_opt=no_namespace --ts_opt=target=web --ts_opt=json_names --ts_opt=explicit_override \
    --ts_opt=runtime_package=npm:google-protobuf --ts_opt=grpc_web_package=npm:grpc-web"

When the service is generated there is still an import like this:

import * as grpc_1 from "@grpc/grpc-js";

and an unused class

export abstract class UnimplementedMyServiceService {
...
}

which is the only construct using grpc_1 types.

Is there any way to remove these constructs from the generation. When I removed them manually all worked properly for me.

Thank you!
Oleg

**Comment by NathanPB** [comment]
Did you found any workarounds?
